VersionControlUI/src/DiffManager.cpp

```diff
@@ -215,18 +233,50 @@ void DiffManager::processNameTextChanges(FilePersistence::IdToChangeDescriptionH
 		if (auto oldNameText = DCast<Model::NameText>(versionNodes.oldNode_))
 			if (auto newNameText = DCast<Model::NameText>(versionNodes.newNode_))
 			{
-				auto id = diffSetup.newVersionManager_->nodeIdMap().idIfExists(newNameText->parent());
-				Q_ASSERT(!id.isNull());
-
-				nameChanges_.insert(oldNameText->get(), {newNameText->get(), id});
+				auto parentId = diffSetup.newVersionManager_->nodeIdMap().idIfExists(newNameText->parent());
+				Q_ASSERT(!parentId.isNull());
 
-				iter = changes.erase(iter);
-				continue;
+				nameChanges_.insert(oldNameText->get(), {newNameText->get(), parentId});
+				nameChangesIdsIsNameText_.insert(id, true);
 			}
 		iter++;
 	}
 }
 
+bool DiffManager::satisfiesTargetNodeIdConstraint(Model::Node* oldNode, Model::Node* newNode,
+																  const DiffSetup& diffSetup)
+{
+	if (targetNodeId_.isNull())
+		return true;
+
+	auto targetOldNode = const_cast<Model::Node*>(diffSetup.oldVersionManager_->nodeIdMap().node(targetNodeId_));
+	auto targetNewNode = const_cast<Model::Node*>(diffSetup.newVersionManager_->nodeIdMap().node(targetNodeId_));
+
+	if ((targetOldNode && targetOldNode->isSameOrAncestorOf(oldNode))
+		 || (targetNewNode && targetNewNode->isSameOrAncestorOf(newNode)))
+		return true;
+
+	return false;
+}
+
+bool DiffManager::satisfiesNameChangeVisualizationConstraints(Model::NodeIdType id)
+__attribute__((optnone))
+{
+	auto iter = nameChangesIdsIsNameText_.find(id);
+
+	if (iter == nameChangesIdsIsNameText_.end())
+		return true;
+
+	if (iter.value() && !nameChangeVisualization_.testFlag(NameText))
+		return false;
+
+	if (!iter.value() && !nameChangeVisualization_.testFlag(References))
```

Comment:
It checks if the id is involved in a name change and iter.value() is used to distinguish wheter it is a NameText or a Reference. If it is e.g. a NameText and the NameText flag is not set, then the method will return false. At the moment, it is used to find out whether the node with the specified id should be visualized or not.